### PR TITLE
feat: apply reviews from #36

### PR DESCRIPTION
**Pull request by jaybxyz**
## Description

This PR applies feedbacks and suggestions from #36.

closes: #36 

## Tasks

- [x] Remove unused function in `app.go` (ref [this comment](https://github.com/tendermint/fundraising/pull/36/files#diff-0f1d2976054440336a576d47a44a37b80cdf6701dd9113012bce0e3c425819b7L617-L625))
- [x] Add if statement to check the auction type in end blocker (ref [this comment](https://github.com/tendermint/fundraising/pull/36/files#diff-0b67bd169bd116f35cb7ed58b15de29fa089f6b1aa074658114b02f18ac055deR37))
- [x] Use `InputOutputCoins` for optimization (ref [this comment](https://github.com/tendermint/fundraising/pull/36/files#diff-c628a857063ee097ccb7b703f42e8f6b6e83cfc1cd9030d36171274500e4fc08R162))
- [x] Add exception logic to block other auction types (ref [this comment](https://github.com/tendermint/fundraising/pull/36/files#diff-25d0c37cfb903f90f50360db69d7857d618f20da32550c210b2e95b798dcb84eR159))
- [x] Update bid grpc query (ref [this comment](https://github.com/tendermint/fundraising/pull/36/files#diff-8f9c6969f818267a8c2ee27072ab8189a3441bbda54debd7edbb8e88585d6bc9R119))

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by jaybxyz** [comment]
@dongsam Can you review this whenever you have time to make sure all TODO lists and suggestions are applied from #36?

File: x/fundraising/keeper/grpc_query.go [hunk 1]
```diff
@@ -115,26 +115,47 @@ func (k Querier) Bids(c context.Context, req *types.QueryBidsRequest) (*types.Qu
...
+	var err error
+
+	store := ctx.KVStore(k.storeKey)
+	switch {
+	case req.Bidder != "" && req.Eligible == "":
+		bids, pageRes, err = queryBidsByBidder(ctx, k, store, req)
+	case req.Bidder == "" && req.Eligible != "":
+		bids, pageRes, err = queryBidsByEligible(ctx, k, store, req)
+	default:
+		bids, pageRes, err = queryAllBids(ctx, k, store, req)
```
Comment (dongsam) [inline x/fundraising/keeper/grpc_query.go]: Wouldn't `case req.Bidder != "" && req.Eligible != ""` be possible if we add a Eligible filter option to `queryBidsByBidder`?

Comment (jaybxyz) [reply x/fundraising/keeper/grpc_query.go]: Oh, I forgot the case of when a user inputs both bidder and eligible params. As you suggested, i think it is possible to add eligible filter option to the function. Let me go add the case.

**Review by dongsam — APPROVED** [review]
LGTM!

**Pull request merged**